### PR TITLE
[Help Wanted] VultureBear: Make it a GlobalBear

### PR DESCRIPTION
**Pull request by Adrianzatreanu**
Fixes https://github.com/coala/coala-bears/issues/817

### CODE REVIEW

**Comment by Adrianzatreanu** [comment]
Help wanted as i have no idea how create a globalbear for a linterbear :D thanks

**Comment by Makman2** [comment]
@Adrianzatreanu you need to do this manually for now. If you want we could work together on GlobalBear support for `@linter` as I'm not sure myself how to design this to work nicely with the current `@linter`^^

**Comment by sils** [comment]
@Makman2 wanna take over?

**Comment by Nosferatul** [comment]
@Adrianzatreanu, @Makman2 would want to help you on this 😄

**Comment by Makman2** [comment]
@Nosferatul cool :) though this one is not easy, global bear support for linters is really some feature... But you can try to get this bear work without `@linter` :+1:

**Comment by Adrianzatreanu** [comment]
im really busy these days, as im leaving to pycon in 2 days, if you wanna do it its all yours, if not, ill ping you once im back :D :p

**Comment by Nosferatul** [comment]
Ok, I'll try for the moment. I believe this is a way to learn more about our bears 😄 .

**Comment by Makman2** [comment]
when you manage this PR, than you are definitely an expert :)

File: bears/python/VultureBear.py [hunk 1]
```diff
@@ -19,6 +15,26 @@ class VultureBear:
...
+        else:
+            return True
+
+    def run(self, filename):
+    """
+    Checks Python code for unused variables and functions using ``vulture``.
+
+    See <https://bitbucket.org/jendrikseipp/vulture> for more information.
+    """
+    self.arguments = self.executable + ' {filename}'
```
Comment (AbdealiJK) [inline bears/python/VultureBear.py]: Do we really want to use string arguments here ? List args are better

File: bears/python/VultureBear.py [hunk 2]
```diff
@@ -19,6 +15,26 @@ class VultureBear:
     ASCIINEMA_URL = 'https://asciinema.org/a/82256'
     CAN_DETECT = {'Unused Code'}
 
-    @staticmethod
-    def create_arguments(filename, file, config_file):
-        return filename,
+    executable = "vulture"
+    arguments = (executable,)
```
Comment (AbdealiJK) [inline bears/python/VultureBear.py]: When you use self.arguments after this you're modifying a class variable. So, that will persist for all objects of the class created after that.

File: bears/python/VultureBear.py [hunk 3]
```diff
@@ -19,6 +15,26 @@ class VultureBear:
...
+    output_regex = re.compile(
+        r'(?P<filename>.*):(?P<line>.*):\s*(?P<message>.*)')
+
+    @classmethod
+    def check_prerequisites(cls):  # pragma: no cover
+        if which("bash") is None:
+            return "bash is not installed."
+        if which("vulture") is None:
```
Comment (AbdealiJK) [inline bears/python/VultureBear.py]: elif ?

File: bears/python/VultureBear.py [hunk 4]
```diff
@@ -19,6 +15,26 @@ class VultureBear:
...
-    def create_arguments(filename, file, config_file):
-        return filename,
+    executable = "vulture"
+    arguments = (executable,)
+    output_regex = re.compile(
+        r'(?P<filename>.*):(?P<line>.*):\s*(?P<message>.*)')
+
+    @classmethod
+    def check_prerequisites(cls):  # pragma: no cover
+        if which("bash") is None:
```
Comment (AbdealiJK) [inline bears/python/VultureBear.py]: Is bash needed ? Why ?

File: bears/python/VultureBear.py [hunk 5]
```diff
@@ -19,6 +15,26 @@ class VultureBear:
...
+            return "bash is not installed."
+        if which("vulture") is None:
+            return ("Vulture is missing. Make sure to install it using "
+                    "``pip3 install happiness``.")
+        else:
+            return True
+
+    def run(self, filename):
+    """
+    Checks Python code for unused variables and functions using ``vulture``.
```
Comment (AbdealiJK) [inline bears/python/VultureBear.py]: Check * not checks ... but not introduced in this PR

File: bears/python/VultureBear.py [hunk 6]
```diff
@@ -19,6 +15,26 @@ class VultureBear:
...
+
+    See <https://bitbucket.org/jendrikseipp/vulture> for more information.
+    """
+    self.arguments = self.executable + ' {filename}'
+    stdout_output, _ = run_shell_command(self.arguments)
```
Comment (AbdealiJK) [inline bears/python/VultureBear.py]: run_shell_command doesn't know the filename to specify ?

File: bears/python/VultureBear.py (hunk 1 above)
Comment (Makman2) [inline bears/python/VultureBear.py]: yes please use list args^^

File: bears/python/VultureBear.py (hunk 3 above)
Comment (sils) [inline bears/python/VultureBear.py]: no, there's a return, if is cleaner IMO but I'd do a newline after every return

File: bears/python/VultureBear.py [hunk 7]
```diff
@@ -19,6 +15,26 @@ class VultureBear:
...
+                    "``pip3 install happiness``.")
+        else:
+            return True
+
+    def run(self, filename):
+    """
+    Checks Python code for unused variables and functions using ``vulture``.
+
+    See <https://bitbucket.org/jendrikseipp/vulture> for more information.
+    """
```
Comment (sils) [inline bears/python/VultureBear.py]: are you seriously indenting this like this? your run function doesn't have any code 📦

**Comment by sils** [comment]
taking over @Adrianzatreanu and @Makman2

**Pull request closed without merging**